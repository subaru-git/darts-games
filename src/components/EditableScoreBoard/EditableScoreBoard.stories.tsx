import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import EditableScoreBoard from './EditableScoreBoard';

export default {
  title: 'Components/EditableScoreBoard',
  component: EditableScoreBoard,
} as ComponentMeta<typeof EditableScoreBoard>;

const Template: ComponentStory<typeof EditableScoreBoard> = (args) => (
  <EditableScoreBoard {...args} />
);

export const Default = Template.bind({});
Default.args = {
  score: [10, 20, 30, 40, 50, 60, 70, 80],
};
