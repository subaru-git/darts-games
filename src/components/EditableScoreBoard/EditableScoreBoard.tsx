import React, { FC, Fragment } from 'react';

type EditableScoreBoardProps = {
  score: number[];
  onChange: (score: number[]) => void;
  setFocus?: (focus: boolean) => void;
};

const EditableScoreBoard: FC<EditableScoreBoardProps> = ({ score, onChange, setFocus }) => {
  return (
    <div className='grid grid-cols-2'>
      <div className='border bg-gray-50 text-center'>Round</div>
      <div className='border bg-gray-50 text-center'>Count</div>
      {score.map((s, i) => (
        <Fragment key={i}>
          <div className='border text-center'>{i + 1}</div>
          <input
            className='border text-center'
            type='number'
            min={0}
            max={180}
            defaultValue={s}
            onChange={(e) => {
              const newScore = [...score];
              newScore[i] = parseInt(e.target.value);
              onChange(newScore);
            }}
            onFocus={() => setFocus?.(true)}
            onBlur={() => setFocus?.(false)}
          />
        </Fragment>
      ))}
    </div>
  );
};

export default EditableScoreBoard;
