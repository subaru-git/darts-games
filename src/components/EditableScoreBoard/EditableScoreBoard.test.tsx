import { render } from '@testing-library/react';
import EditableScoreBoard from './EditableScoreBoard';

test('should rendering', () => {
  const onChange = jest.fn();
  const { container } = render(<EditableScoreBoard score={[]} onChange={onChange} />);
  expect(container).toMatchSnapshot();
});
