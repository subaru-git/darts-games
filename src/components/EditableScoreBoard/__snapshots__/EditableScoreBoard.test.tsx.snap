// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="grid grid-cols-2"
  >
    <div
      class="border bg-gray-50 text-center"
    >
      Round
    </div>
    <div
      class="border bg-gray-50 text-center"
    >
      Count
    </div>
  </div>
</div>
`;
